types/compilation/compilation.interfaces.ts

```diff
@@ -30,6 +30,7 @@ export type CompilationResult = {
     buildResult?: unknown;
     inputFilename?: string;
     asm?: ResultLine[];
+    asmSize?: number;
```

Comment:
This was missing in the interface and the base-compiler change was complaining. Note that I've found more issues with this file (Most notably that `asm` proper type should be `asm?: ResultLine[] | string`, but this change is outside the scope of this PR, will create another for that